feat: Added docs for load balancing with Varnish.

ritsingh opened the pull request:
Details:
Added detailed steps to be followed while configuring Varnish for load balancing.

Commit:
feat: Added docs for load balancing with Varnish.

Details:
Added detailed steps to be followed while configuring Varnish for load balancing.

arobson commented on tutorials/load-balancing-with-varnish.md:
````diff
@@ -0,0 +1,91 @@
+ }
+ ```
+
+This defines a backend in Varnish called `default`. When needs to get content from this backend it will connect to port 8080 on `localhost (127.0.0.1)`.
````
Minor correction: "When Varnish serves content from this backend, it will connect to port 8080 on `127.0.0.1`."

arobson commented on tutorials/load-balancing-with-varnish.md:
```diff
@@ -0,0 +1,91 @@
+  set req.http.X-Authorization = req.http.Authorization;
+  unset req.http.Authorization;
+  unset req.http.If-Modified-Since;
+
```
Is there maybe an initial, opening `if` missing here? Based on the description of the logic above and indentation level here, it looks like there should be an opening conditional.

ritsingh replied:
I have added this section from the link that Benjamin provided, you can look over here https://gist.githubusercontent.com/bcoe/1a80cce763c8ad50fd1ace543f6f756c/raw/7d4f5d0cabefeb51572776fc6fd866f04ff0b681/varnish.vcl

arobson replied:
@ritsingh - right, if you look at that link, there is an opening if that I think you're missing:

`if (req.url ~ "@" || req.url ~ "osc-cli" || req.url ~ "feature-toggle") {`

@bcoe - would you take a look at this and let me know if I'm reading this wrong? I'm certainly not a Varnish expert and I could be incorrect about how it's being used in conjunction with npme here, but I think the additional opening `if` is needed.

Commit:
feat: Updated docs for load balancing with Varnish.

Details:
Modified "When needs to get content from this backend it will connect to port 8080 on `localhost (127.0.0.1)`." to "When Varnish serves content from this backend, it will connect to port 8080 on `127.0.0.1`."

Commit:
feat: Updated docs for 'Load balancing with Varnish'.

Details:
Added check constraint for scoped packages.

mahikhond committed:
Merge branch 'master' into load-balancing-with-varnish
